Neos.ContentGraph.DoctrineDbalAdapter/src/NodeQueryBuilder.php

```diff
@@ -70,6 +68,16 @@ public function buildChildNodeAggregateQuery(NodeAggregateId $parentNodeAggregat
             ]);
     }
 
+    public function buildParentNodeAggregateQuery(): QueryBuilder
+    {
+        return $this->createQueryBuilder()
+            ->select('n.*, h.contentstreamid, h.subtreetags, dsp.dimensionspacepoint AS covereddimensionspacepoint')
+            ->from($this->tableNames->node(), 'n')
+            ->innerJoin('n', $this->tableNames->hierarchyRelation(), 'h', 'h.parentnodeanchor = n.relationanchorpoint')
```

Comment:
fyi the only difference to the previously used `buildBasicNodeAggregateQuery` is that we join on `h.parentnodeanchor` instead of `h.childnodeanchor`

I had the feeling introducing this might deduplicate code another place and it could:

In `findParentNodeAggregateByChildOriginDimensionSpacePoint` we use the same logic besides one additional `where`. So i think it could be nicely deduplicated like:

```php
public function findParentNodeAggregateByChildOriginDimensionSpacePoint(NodeAggregateId $childNodeAggregateId, OriginDimensionSpacePoint $childOriginDimensionSpacePoint): ?NodeAggregate
    {
        $subQueryBuilder = ...;
    
        $queryBuilder = $this->nodeQueryBuilder->buildParentNodeAggregateQuery()
            ->andWhere('n.nodeaggregateid = (' . $subQueryBuilder->getSQL() . ')')
            ->setParameters([
                'contentStreamId' => $this->contentStreamId->value,
                'childNodeAggregateId' => $childNodeAggregateId->value,
                'childOriginDimensionSpacePointHash' => $childOriginDimensionSpacePoint->hash,
            ]);
```


Also i dont exactly understand how this utility is supposed to work (when do we pass arguments or when to use `setParameters`) but because we dont return a final query i think its `buildBasicParentNodeAggregateQuery`

Reply:
Is you comment only for clarification? Or do you complain about the partially duplication of the queries and want to get that changed?

Reply:
i was just thoroughly reviewing this and partly investigating and trying to understand. So i guess 50/50 ill create a followup pr :D